src/Compiler/Microsoft.CodeAnalysis.Razor.Compiler/src/SourceGenerators/RazorSourceGenerator.RazorProviders.cs

```diff
@@ -42,7 +44,9 @@ public partial class RazorSourceGenerator
                 razorLanguageVersion = RazorLanguageVersion.Latest;
             }
 
-            var razorConfiguration = new RazorConfiguration(razorLanguageVersion, configurationName ?? "default", Extensions: [], UseConsolidatedMvcViews: true);
+            var isComponentParameterSupported = CSharpCompilation.Create("components", references: references).HasAddComponentParameter();
```

Comment:
🤔 Is there a better way to do this? It feels unnecessarily expensive to create a temporary `CSharpCompilation` with all of the metadata references just to get the value of this option. Could the compilation be passed to `ComputeRazorSourceGeneratorOptions` rather than the metadata references? Also, the old code checked the metadata references for `Microsoft.AspNetCore.Components.dll` before creating the compilation. Is that optimization no longer needed?
 
@chsienki, do you have any concerns with this compilation being created here? Note that it happens much earlier in `RazorSourceGenerator.Initialize(...)` than it did before. I'm definitely not an expert on the incremental generator machinery, so I'm not sure what kind of impact this will have.

An alternative approach might be to make the `RazorSourceGenerationOptions.Configuration` lazily computed, or just create the `RazorConfiguration` in `GetDeclarationProjectEngine(...)` and `GetGenerationProjectEngine(...)` and remove it from `RazorSourceGenerationOptions` altogether.

Reply:
@DustinCampbell I wrote this, lol.

Agree that we should filter to MS.ANC.Components.dll to make it faster, that's a bug on me porting it over. 

It does seem wild that we do this, but as its structured it's still the fastest way to do this. The compilation is going to change continuously, which means we'd end up re-running this code extremely frequently. The metadata refs change very infrequently to be amortized to the point where this basically only runs once (especially if we're actually filtering the reference list).

Now, that being said we could probably split this up into two steps, one of which runs from the compilation, so the rest doesn't run. That would be faster for batch-compilation but I can't predict the difference in IDE without actually benchmarking it (I suspect its probably a wash though).

I have it on my TODO list to clean some of this up. Let me open an issue to track that so we don't forget about it.